dessia_common/document_generator.py

```diff
@@ -0,0 +1,93 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+""" Write document file. """
+from typing import List, Tuple
+
+import docx
+
+
+class DocxWriter:
+    """ write a docx file. """
+
+    def __init__(self, filename: str):
+        self.filename = filename
+        self.doc = docx.Document()
+        self.headings = None
+
+    def add_headings(self, headings: List[Tuple[str, int]]) -> 'DocxWriter':
+        """ Add a list of headings to the document. """
+        new_doc = self.doc
+        for heading in headings:
+            new_doc.add_heading(heading[0], heading[1])
+        new_writer = DocxWriter(self.filename)
+        new_writer.doc = new_doc
+        new_writer.headings = headings
+        new_writer.add_page_breaks(1)
+        return new_writer
+
+    def add_paragraphs(self, paragraphs: List[str]) -> 'DocxWriter':
+        """ Add a list of paragraphs to the document. """
+        new_doc = self.doc
```

Comment:
Why do you need to assign this variable ? Is this in order to assign a temporary variable ? it feels weird to not just use "self.doc" in the method

Reply:
it's just so as not to modify the self directly